### DISCUSSION TITLE
"404 This page could not be found" in dynamic page

### DISCUSSION THREAD

**Discussion by reposman33** [vercel/next.js#24279] [Help] [answered]
Hi,

I am working through the Learn Next.js tutorials. I am at [Dynamic Routes](https://nextjs.org/learn/basics/dynamic-routes/implement-getstaticprops)
I want to create a dynamic route for the markdown files.

First I added getStaticProps to the index.js file to retrieve a listing of all markdown filenames in the post directory. The listing is used to create a series of Links on the landing page (/). That works: clicking a Link brings me to a route with the fileName slug: e.g. 

`localhost:3000/ssg-ssr`

But the response of the dynamic route is `404 This page could not be found.`

- I added getStaticPaths that retrieves the fileNames;
- I added getStaticProps that retrieves the data for the file corresponding to current route
- I added the parameter to the default Component to render the retrieved data.

I checked my code against the link given at the end in `(If it’s still not working) The remaining code should look like this.`

I guess I'm missing something here...?

Repos: [https://github.com/reposman33/nextjs.blog.git](https://github.com/reposman33/nextjs.blog.git)

**Comment by gomflo** [comment, marked as answer]
File `index.js` from this:

```js
<Link href={`/${id}`}>
  <a href={`/${id}`}>{id}</a>
</Link>
```

to this: 
```js
<Link href={`/posts/${id}`}>
  <a>{id}</a>
</Link>
```

When your using Link component is not need to add href to the html tag.

In `posts/[id].js` i did minimal edits **just to make it work** from this:

```js
const getStaticPaths = async () => {
    const paths = getAllPostIds();
    return { paths, fallback: false };
};
```

To this: 

```js
const getStaticPaths = async () => {
    const paths = getAllPostIds();
    return { paths: [], fallback: 'blocking' };
};
```

Function `getAllPostIds` must return an array of paths.

About the fallback option: https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation

**Reply by reposman33** [reply]
Thanks, the return value of getAllPostIds() was indeed the problem. It was not in the required format [https://nextjs.org/learn/basics/dynamic-routes/implement-getstaticpaths](https://nextjs.org/learn/basics/dynamic-routes/implement-getstaticpaths)

**Comment by haobk57** [comment]
you try to modify the path of postsDirectory variable in posts.js to: 
`const postsDirectory = path.join(process.cwd(), 'pages/posts')`
